guava/src/com/google/common/collect/CollectSpliterators.java

```diff
@@ -96,7 +96,7 @@ public int characteristics() {
       @CheckForNull
       public Comparator<? super T> getComparator() {
         if (hasCharacteristics(Spliterator.SORTED)) {
-          return comparator;
+          return Ordering.natural().equals(comparator) ? null : comparator;
```

Comment:
We should special-case Comparator.naturalOrder() too.

Reply:
Okay I've updated it to handle `Comparator.naturalOrder()` as well.